### ISSUE TITLE
Libdoc: Drop `typing.` prefix from type hints originating from the `typing` module

### ISSUE THREAD

**Issue by Snooz82** [robotframework/robotframework#3724]
Type hints from typing looks a bit uncomfortable ...

```python
from typing import Optional, Union, Any

def keyword(whatever: Any, stringy: str = 'some', maybe: Optional[str] = None, num: Union[int, str] = 2, )
    pass
```

```
whatever: typing.Any,
stringy: str = some,
maybe: typing.Optional[str] = None,
num: typing.Union[int, str] = 2
```

That should be something like that:
```
whatever: Any,
stringy = some,
maybe: Optional[str] = None,
num: Union[int, str] = 2
```

**Comment by pekkaklarck** [comment]
#3748 proposes storing information about Unions separately in specs and showing that in HTML like `<int> or <float>`. That would solve this problem with Union and because Optional is actually a shortcut for a Union with None also with that.

An easy solution for Any is not showing that information at all. There's only a semantic difference between `def kw(arg: Any)` and `def kw(arg)` anyway and it may be best not to bother users with that.

**Comment by MaciejWiczk** [comment]
@Snooz82, don't You think that `stringy: str = some` have better meaning than `stringy = some` ? I can imagine a situation where this argument would have value None by default, and would be hard o catch what kind of variable it should be.

**Comment by pekkaklarck** [comment]
This has been already implemented as part of PR #3771 but needs a test still.
